### PR TITLE
[Merged by Bors] - Added config::merge::chainable_merge()

### PR DESCRIPTION
**Pull request by razvan**

### CODE REVIEW

File: src/config/merge.rs [hunk 1]
```diff
@@ -391,4 +418,24 @@ mod tests {
             BTreeMap::from([("a", Acc(4)), ("b", Acc(2)), ("c", Acc(5))])
         );
     }
+
+    #[test]
+    fn test_chainable_merge() {
```
Comment (teozkr) [inline src/config/merge.rs:423]: No need to keep the same test case as both a regular unit test and a doctest

**Review by teozkr — COMMENTED** [review]
Did you upload your branch where you use this in the operator?

File: src/config/merge.rs (hunk 1 above)
Comment (razvan) [reply src/config/merge.rs:423]: Here is a usecase: https://github.com/stackabletech/hdfs-operator/pull/164/files#diff-b10253ea28ef51076309f375e97cd5e2795500b0e07c5326c413055bffcb4a85R253

I will remove the test

**Review by maltesander — APPROVED** [review]
LGTM.

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**